src/software/jetson_nano/thunderloop.cpp

```diff
@@ -96,41 +45,38 @@ void Thunderloop::runLoop()
 {
     // Timing
     struct timespec next_shot;
-    struct timespec start_time;
-    struct timespec end_time;
     struct timespec time;
 
-#define TIME_EXPRESSION(arg)                                                             \
-    clock_gettime(0, &start_time);                                                       \
-    arg;                                                                                 \
-    clock_gettime(0, &end_time);                                                         \
-    timespec_diff(&end_time, &start_time, &time);
-
     // Input buffer
     TbotsProto::PrimitiveSet new_primitive_set;
     TbotsProto::Vision new_vision;
 
     // Loop interval
-    int interval = 1 / loop_hz_ * static_cast<int>(NANOSECONDS_PER_SECOND);
+    int interval =
+        static_cast<int>(1.0f / static_cast<float>(loop_hz_) * NANOSECONDS_PER_SECOND);
 
     // Start the services
     motor_service_->start();
 
     // Get current time
-    clock_gettime(0, &next_shot);
+    // Note: CLOCK_MONOTONIC is used over CLOCK_REALTIME since CLOCK_REALTIME can jump
+    // backwards
+    clock_gettime(CLOCK_MONOTONIC, &next_shot);
 
     // Start after one second
     next_shot.tv_sec++;
 
     for (;;)
     {
         // Wait until next shot
-        clock_nanosleep(0, TIMER_ABSTIME, &next_shot, NULL);
+        // Note: CLOCK_MONOTONIC is used over CLOCK_REALTIME since CLOCK_REALTIME can jump
+        // backwards
+        clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &next_shot, NULL);
 
         // Poll network service and grab most recent messages
-        TIME_EXPRESSION(std::tie(new_primitive_set, new_vision) =
-                            network_service_->poll(robot_status_));
-        thunderloop_status_.set_network_service_poll_time_ns(time.tv_nsec);
+        auto [new_primitive_set, new_vision] = network_service_->poll(robot_status_);
+        thunderloop_status_.set_network_service_poll_time_ns(
+            static_cast<unsigned long>(time.tv_nsec));
```

Comment:
why aren't the seconds being added? Are we _that_ sure that our code won't take more than a second to run?

Reply:
yeah thats a good point, we are pretty sure, but it would be bad if something stalled

added a check on loop timing that makes sure we fit inside 1 period